### PR TITLE
Fe nc news view single article

### PR DESCRIPTION
**Pull request by alicjafleszar**

### CODE REVIEW

File: src/components/ArticlesSection/ArticlesSection.jsx [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+  const [isLoading, setIsLoading] = useState(false)
+  
+  useEffect(() => {
+    setIsLoading(true)
+    fetchArticles()
+      .then(articles => {
+        setArticles(articles)
+        setIsLoading(false)
+      })
+  }, [])
```
Comment (Rayhaan2k9) [inline src/components/ArticlesSection/ArticlesSection.jsx]: This whole component isn't really needed as ll the behaviour can be implemented in the ArticlesList component. However, you can still keep it if you like it like this 🙂

**Review by Rayhaan2k9 — APPROVED** [review]
Brilliant work, Alicja. Merge away! 😄

File: src/components/MainContent/MainContent.jsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 import ArticlesSection from "../ArticlesSection/ArticlesSection"
+import SingleArticle from "../SingleArticle/SingleArticle"
 
 const MainContent = () => {
   return (
     <Routes>
       <Route path="/" element={<ArticlesSection />} />
       <Route path="/articles" element={<ArticlesSection />} />
+      <Route path="/:article" element={<SingleArticle />} />
     </Routes>
```
Comment (EddTheDuck) [inline src/components/MainContent/MainContent.jsx]: The URL should be /articles/:article_id.

If you were to add another route below path="/:article" you would never reach it as what you have is a parametric endpoint.

File: src/components/SingleArticle/SingleArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+import { useLocation, useParams } from "react-router-dom"
+import { fetchArticleById } from "../../api/api";
+import ArticleSection from "../ArticleSection/ArticleSection";
+
+function SingleArticle() {
+  const [isLoading, setIsLoading] = useState(false)
+  const [singleArticle, setSingleArticle] = useState({})
+  const { article } = useParams();
+  const articleId = article.split('-').slice(-1)
```
Comment (EddTheDuck) [inline src/components/SingleArticle/SingleArticle.jsx]: This is an odd way to get the articleId - read the notes on useParams and react router to refresh your memory.

File: src/components/ArticlesList/ArticlesList.jsx [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
-      })
-  }, [])
+  // useEffect(() => {
+  //   setIsLoading(true)
+  //   fetchArticles()
+  //     .then(articles => {
+  //       setArticles(articles)
+  //       setIsLoading(false)
+  //     })
+  // }, [])
```
Comment (EddTheDuck) [inline src/components/ArticlesList/ArticlesList.jsx]: Do not leave commented out code when submitting a PR.

File: src/components/ArticleCard/ArticleCard.jsx [hunk 5]
```diff
@@ -1,17 +1,23 @@
+import { Link } from "react-router-dom"
 import './ArticleCard.css'
-const ArticleCard = ({ title, topic, created_at, article_img_url, author }) => {
+
+const ArticleCard = ({ id, title, topic, created_at, article_img_url, author }) => {
   const dateString = new Date(created_at).toLocaleString('default', {month: 'short', day: 'numeric', year: 'numeric'})
 
+  const articlePath = `/${encodeURIComponent(title.toLowerCase().replaceAll(' ', '-'))}-${id}`
```
Comment (EddTheDuck) [inline src/components/ArticleCard/ArticleCard.jsx]: I think this is very unnessary and needs revision.

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
I think because you have set your path as a parametric endpoint you have tried to come up with a work around.

It currently is not functioning as it should be. Please go read the NCNotes on React Router and refactor. 

If you are still struggling submit an NC-Help.

File: src/components/ArticleCard/ArticleCard.jsx (hunk 5 above)
Comment (alicjafleszar) [reply src/components/ArticleCard/ArticleCard.jsx]: I feel like including the titile of the article in the URL makes it clearer for the user what site they are entering and what content they're about to see before clicking the link if it was let's say, shared with them. Would separating article_id from title with a slash be a better way to do it?
```suggestion
  const articlePath = `/articles/${topic}/${id}/${encodeURIComponent(title.toLowerCase().replaceAll(/[ .]/g, '-'))}`
```

File: src/components/ArticlesSection/ArticlesSection.jsx (hunk 1 above)
Comment (alicjafleszar) [reply src/components/ArticlesSection/ArticlesSection.jsx]: Thanks :)

File: src/components/MainContent/MainContent.jsx (hunk 2 above)
Comment (alicjafleszar) [reply src/components/MainContent/MainContent.jsx]: Thanks for pointing it out! In this case, this is how I would refactor it, if you agree that the title part of URL can stay.
```suggestion
        <Route path="/articles/:topic/:article_id/:article_title" element={<SingleArticle />} />
    </Routes>
```

File: src/components/SingleArticle/SingleArticle.jsx (hunk 3 above)
Comment (alicjafleszar) [reply src/components/SingleArticle/SingleArticle.jsx]: That was my workaround to extract article ID from the URL which didn't include separated ID. I'll refactor this part. Is there something else in this section that I should improve or was that the only thing?

**Pull request merged**